testplan/testing/base.py

```diff
@@ -838,6 +839,6 @@ def run_testcases_iter(self, testsuite_pattern="*", testcase_pattern="*"):
     def test_command_filter(self, testsuite_pattern, testcase_pattern):
         """
         Return the base test command with additional filtering to run a
-        specific set of testcases.
+        specific set of testcases. To be implemented by concrete subclasses.
         """
         raise NotImplementedError
```

Comment:
This is in line with what's present in other places in this codebase, but just for the record methods like this ought to be marked `abstractmethod` since normal IDEs will highlight to the user that they need to implement the method.

That however will require a broad refactor of several base classes and so is outside the scope of this PR.